### PR TITLE
Wip/batch inference

### PR DESCRIPTION
**Pull request by chrise96**
Batch inference implementation using build_model and DataLoader.

ContainerDataset(Dataset) class is partly based on DefaultPredictor https://github.com/facebookresearch/detectron2/blob/main/detectron2/engine/defaults.py#L252 

Results are saved directly to database instead of blob storage -> database

To test the detection results locally, please clone this commit: d3428a07dc5fd354cdb30c3da672d257f695545a 

TODO: implement select query to database in postprocessing.py

### CODE REVIEW

File: detection.Dockerfile [hunk 1]
```diff
@@ -28,10 +29,10 @@ RUN python -m pip install -e detectron2
 
 WORKDIR /app
 COPY configs/* /app/configs/
-COPY inference.py evaluation.py utils_coco.py /app/
+COPY inference_batch.py evaluation.py utils_coco.py upload_to_postgres.py /app/
 COPY utils /app/utils
 # copy weights
-COPY model_final.pth /app/
+# COPY model_final.pth /app
```
Comment (epureanudiana) [inline detection.Dockerfile:35]: this is uncommented in the end, right?

File: inference_batch.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+            blob_name=f"{start_date_dag}/{results_file_name}",
+            local_file_path=results_file_name,
+        )
+
+    if data_results:
+        print("Inserting data into database...")
+        table_name = "detections"
+
+        # Get columns
+        sql = f"SELECT * FROM {table_name} LIMIT 0"
```
Comment (epureanudiana) [inline inference_batch.py:206]: what does limit 0 mean?

Comment (chrise96) [reply inference_batch.py:206]: Dont return the values, only get the structure of the database table

File: detection.Dockerfile (hunk 1 above)
Comment (chrise96) [reply detection.Dockerfile:35]: Oeps, idd

**Pull request merged**